expression/builtin_miscellaneous.go

```diff
@@ -1043,5 +1044,38 @@ type uuidShortFunctionClass struct {
 }
 
 func (c *uuidShortFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "UUID_SHORT")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	bf, err := newBaseBuiltinFuncWithTp(ctx, c.funcName, args, types.ETInt)
+	if err != nil {
+		return nil, err
+	}
+	bf.tp.Charset, bf.tp.Collate = ctx.GetSessionVars().GetCharsetInfo()
+	bf.tp.Flen = 36
+	sig := &builtinUUIDShortSig{bf}
+	return sig, nil
+}
+
+type builtinUUIDShortSig struct {
+	baseBuiltinFunc
+}
+
+func (b *builtinUUIDShortSig) Clone() builtinFunc {
+	newSig := &builtinUUIDShortSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+// evalInt evals a builtinUUIDShortSig.
+// See https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid-short
+func (b *builtinUUIDShortSig) EvalInt(_ chunk.Row) (d int64, isNull bool, err error) {
+	buf := make([]byte, 2, 8)
+	buf = append(buf, uuid.NodeID()...)
+
+	var uuid int64
+	for _, b := range buf {
+		uuid = (uuid << 8) | int64(b)
+	}
```

Comment:
it is the same as `uuid := binary.BigEndian.Uint64(buf)`

you can see that https://play.studygolang.com/p/rawBAlWK_yv